### PR TITLE
Added basic Pylint analyzers/checkers to Concept Exercises 

### PR DESCRIPTION
**Pull request by BethanyG**
Added base `analyzer.py` files for `card-games`, `cater-waiter`, and `chaitanas-colossal-coaster`.  
This will enable them to have a Pylint check as basic analysis while we get more custom analysis written.

### CODE REVIEW

File: lib/card-games/analyzer.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+"""
+Basic analyzer for the `card-games` exercise.
+Only has pylint checks active currently.
```
Comment (kotp) [inline lib/card-games/analyzer.py:3]:
```suggestion
Only has pylint checks are active currently.
```

File: lib/card-games/analyzer.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+class Comments(BaseFeedback):
+    NO_MODULE = ("general", "no_module")
+    NO_METHOD = ("two-fer", "no_method")
+    MALFORMED_CODE = ("general", "malformed_code")
+
+def analyze(in_path: Path, out_path: Path):
+    """
+    Analyze the user's Two Fer solution to give feedback. Outputs a JSON that
```
Comment (kotp) [inline lib/card-games/analyzer.py:22]:
```suggestion
    Analyze the user's Two Fer solution to give feedback. Outputs JSON that
```

File: lib/card-games/analyzer.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        comments.append(Comment(type=CommentTypes.ESSENTIAL, params={}, comment=Comments.MALFORMED_CODE))
+    finally:
+        if comments:
+            return Analysis.require(comments).dump(output_file)
+
+
+    # Generate PyLint comments for additional feedback.
+    comments.extend(generate_pylint_comments(in_path))
+
+
```
Comment (kotp) [inline lib/card-games/analyzer.py:57]:
```suggestion
```

Not sure if 3 blank lines is idiomaticaly placed here, but…

File: lib/cater-waiter/analyzer.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+"""
+import ast
+from pylint import epylint as lint
+from pathlib import Path
+
+from common import Analysis, BaseFeedback, Summary
+from common.comment import Comment, CommentTypes
+from common.pylint_comments import generate_pylint_comments
+
+
```
Comment (kotp) [inline lib/cater-waiter/analyzer.py:13]:
```suggestion
```

File: lib/cater-waiter/analyzer.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    # input file - if it can't be found, fail and bail
+    try:
+        user_solution = in_path.read_text()
+    except OSError as err:
+        # fail out fast with an ESSENTIAL (required) type comment for the student
+        comments.append(Comment(type=CommentTypes.ESSENTIAL, params={}, comment=Comments.MALFORMED_CODE))
+    finally:
+        if comments:
+            return Analysis.require(comments).dump(output_file)
+
```
Comment (kotp) [inline lib/cater-waiter/analyzer.py:41]:
```suggestion
```

File: lib/cater-waiter/analyzer.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    # AST - if an AST can't be made, fail and bail
+    try:
+        tree = ast.parse(user_solution)
+    except Exception:
+        # If ast.parse fails, assume malformed code and fail with an ESSENTIAL (required) type comment for the student
+        comments.append(Comment(type=CommentTypes.ESSENTIAL, params={}, comment=Comments.MALFORMED_CODE))
+    finally:
+        if comments:
+            return Analysis.require(comments).dump(output_file)
+
```
Comment (kotp) [inline lib/cater-waiter/analyzer.py:52]:
```suggestion
```

File: lib/chaitanas-colossal-coaster/analyzer.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    # input file - if it can't be found, fail and bail
+    try:
+        user_solution = in_path.read_text()
+    except OSError as err:
+        # fail out fast with an ESSENTIAL (required) type comment for the student
+        comments.append(Comment(type=CommentTypes.ESSENTIAL, params={}, comment=Comments.MALFORMED_CODE))
+    finally:
+        if comments:
+            return Analysis.require(comments).dump(output_file)
+
```
Comment (kotp) [inline lib/chaitanas-colossal-coaster/analyzer.py:41]:
```suggestion
```

File: lib/chaitanas-colossal-coaster/analyzer.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+    # Generate PyLint comments for additional feedback.
+    comments.extend(generate_pylint_comments(in_path))
+
+
```
Comment (kotp) [inline lib/chaitanas-colossal-coaster/analyzer.py:57]:
```suggestion
```

File: lib/chaitanas-colossal-coaster/analyzer.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    # AST - if an AST can't be made, fail and bail
+    try:
+        tree = ast.parse(user_solution)
+    except Exception:
+        # If ast.parse fails, assume malformed code and fail with an ESSENTIAL (required) type comment for the student
+        comments.append(Comment(type=CommentTypes.ESSENTIAL, params={}, comment=Comments.MALFORMED_CODE))
+    finally:
+        if comments:
+            return Analysis.require(comments).dump(output_file)
+
```
Comment (kotp) [inline lib/chaitanas-colossal-coaster/analyzer.py:52]:
```suggestion
```

File: lib/chaitanas-colossal-coaster/analyzer.py [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+"""
+import ast
+from pylint import epylint as lint
+from pathlib import Path
+
+from common import Analysis, BaseFeedback, Summary
+from common.comment import Comment, CommentTypes
+from common.pylint_comments import generate_pylint_comments
+
+
```
Comment (kotp) [inline lib/chaitanas-colossal-coaster/analyzer.py:13]:
```suggestion
```

File: lib/cater-waiter/analyzer.py [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+    # Generate PyLint comments for additional feedback.
+    comments.extend(generate_pylint_comments(in_path))
+
+
```
Comment (kotp) [inline lib/cater-waiter/analyzer.py:57]:
```suggestion
```

File: lib/card-games/analyzer.py [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    # input file - if it can't be found, fail and bail
+    try:
+        user_solution = in_path.read_text()
+    except OSError as err:
+        # fail out fast with an ESSENTIAL (required) type comment for the student
+        comments.append(Comment(type=CommentTypes.ESSENTIAL, params={}, comment=Comments.MALFORMED_CODE))
+    finally:
+        if comments:
+            return Analysis.require(comments).dump(output_file)
+
```
Comment (kotp) [inline lib/card-games/analyzer.py:41]:
```suggestion
```

File: lib/card-games/analyzer.py [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+"""
+import ast
+from pylint import epylint as lint
+from pathlib import Path
+
+from common import Analysis, BaseFeedback, Summary
+from common.comment import Comment, CommentTypes
+from common.pylint_comments import generate_pylint_comments
+
+
```
Comment (kotp) [inline lib/card-games/analyzer.py:13]:
```suggestion
```

**Review by kotp — COMMENTED** [review]
One grammatical fix, unsure about vertical space changes.

File: lib/card-games/analyzer.py (hunk 1 above)
Comment (BethanyG) [reply lib/card-games/analyzer.py:3]: Would counter suggest "Only pylint checks are active currently."  As the `has` doesn't make sense.

Comment (BethanyG) [reply lib/card-games/analyzer.py:3]:
```suggestion
Only Pylint checks are active currently.
```

**Review by kotp — APPROVED** [review]

**Pull request merged**